_posts/2018-08-28-announcing-riff-0-1-2-on-Knative.md

```diff
@@ -0,0 +1,197 @@
+```
+
+## function chaining over channels
+
+[Knative/eventing](https://github.com/knative/eventing) now includes support for subscriptions to [send function responses](https://github.com/knative/eventing/pull/325) to another channel. This feature can be used through the riff CLI by specifying the output channel with `--output` or `-o` when creating a subscription.
+
+### 1. create a simple hello function
+Start with 2 files in a directory.
+
+#### hello.js  
+```js
+module.exports = x => {
+  var out = 'hello ' + x
+  console.log(out)
+  return out
+}
+```
+
+#### Dockerfile  
+```dockerfile
+FROM projectriff/node-function-invoker:0.0.8
+ENV FUNCTION_URI /functions/hello.js
+ADD hello.js ${FUNCTION_URI}
+```
+
+Now build the function into a container image, and use it create a Knative Service. The `dev.local` prefix tells Knative to use the image from the local docker daemon instead of pulling an image from a remote container registry.
+```sh
+docker build -t dev.local/hello:v1 .
+riff service create hello --image dev.local/hello:v1
+```
+> To make this work in a hosted environment like GKE, change the names of the images to prefix them with your own registry, and push the images there before calling riff service create.
+
+Using a tool like kail makes it easy to watch the function container log in a separate terminal window.
+```sh
+kail -d hello-00001-deployment -c user-container
+```
+
+When you invoke hello, kail relays the container log.
+```sh
+riff service invoke hello -- -HContent-Type:text/plain '-w\n' -d hello
+```
+
+#### kail output
+```
+...[user-container]: hello hello
+```
+
+### 2. create a square function
+Follow the same pattern to create a square function.
+
+#### square.js  
+```js
+module.exports = x => {
+  var out = x**2
+  console.log('%s**2 = %s', x, out)
+  return out
+}
+```
+
+#### Dockerfile  
+```dockerfile
+FROM projectriff/node-function-invoker:0.0.8
+ENV FUNCTION_URI /functions/square.js
+ADD square.js ${FUNCTION_URI}
+```
+
+Build the image and create the function
+```sh
+docker build -t dev.local/square:v1 .
+riff service create square --image dev.local/square:v1
+```
+
+Invoking square with a number should return the number squared.
+```sh
+riff service invoke square -- -HContent-Type:text/plain '-w\n' -d 7
+```
+```sh
+curl...
+49
+```
+
+### 3. create a function for generating random numbers
+This function is a little more complex. To help keep this blog post short, we have published a _random_ function on [dockerhub](https://hub.docker.com/r/jldec/random/tags/). The source can be found on [GitHub](https://github.com/jldec/random).
+
+Create the function using the image from dockerhub.
+```sh
+riff service create random --image jldec/random:v0.0.1
+```
+
+Invoke the function to send posts to square.
+```sh
+riff service invoke random -- -H 'Content-Type:application/json' -d '{"url":"http://square.default.svc.cluster.local"}'
+```
+
+Tailing the log of the square function should show repeated squaring
```

Comment:
Huh. How are the random numbers being sent to the square function?

Reply:
May be worth pointing that out in the text for anyone out there who, like me, doesn't notice the url parameter.